0015-ilp-addresses/0015-ilp-addresses.md

```diff
@@ -0,0 +1,123 @@
+2. Each "segment" of the address MUST contain one or more of the following characters:
+    - Alphanumeric characters, upper or lower case. (Addresses are **case-sensitive** so that they can contain data encoded in formats such as base64url.)
+    - Underscore (`_`)
+    - Tilde (`~`)
+    - Hyphen (`-`)
+3. Each segment MUST be separated from other segments by a period character (`.`).
+4. Prefix addresses MUST end in a period (`.`) character and MAY contain any number of segments after the allocation scheme prefix.
+6. Destination addresses MUST NOT end in a period (`.`) character, and MUST contain at least two segments after the allocation scheme prefix.
+
+The following regular expressions summarize these requirements:
+
+| Address Type        | Regular Expression                         |
+|:--------------------|:-------------------------------------------|
+| All addresses       | `^g\.[a-zA-Z0-9._~-]*$`                    |
+| Prefix address      | `^g\.([a-zA-Z0-9_~-]+\.)*$`                |
+| Destination address | `^g\.([a-zA-Z0-9_~-]+\.)+[a-zA-Z0-9_~-]+$` |
+
+## Example Addresses
+
+`g.acme.bob` - a destination address to the account "bob" in the ledger "acme".
+
+`g.us.fed.ach.0.acmebank.swx0a0.acmecorp.sales.199.cdfa5e16-e759-4ba3-88f6-8b9dc83c1868.2` - destination address for a particular invoice, which can break down as follows:
+
+- Neighborhoods: `us`, `fed`, `ach`, `0`, `acmebank`
+- Ledger: `swx0a0`
+- Subledger: `acmecorp`
+- Account group: `sales`
+- Account: `199`
+- Interactions: `cdfa5e16-e759-4ba3-88f6-8b9dc83c1868`, `2`
+
+`g.` - the shortest possible prefix address. All entries that are in the general allocation scheme.
+
+`g.crypto.bitcoin.` - prefix address for the public Bitcoin blockchain
+
+## General Allocation Scheme
+
+The general allocation scheme for ILP Addresses is currently the only supported scheme. It uses the prefix `g.`.
+
+This scheme has no central issuing authority or mechanism, so more than one entity can use the same address. In this case, some connectors may prepare a route to a different account than intended. In this failure case, no money moves because the receiver does not send the fulfillment. Participants in the interledger can reduce the chances of encountering this failure case by choosing addresses carefully and by properly managing connectors' routing tables.
+
+To make routing work well, we recommend including the following components as segments in an address (in order):
+
+- [Neighborhoods](#neighborhoods)
+- [Payment rail](#payment-rail)
+- [Ledger prefix](#ledger-prefix)
+- [Subledger and account group](#subledger-and-account-group)
+- [Account identifier](#account-identifier)
+- [Interactions](#interactions)
+
+
+### Neighborhoods
+
+Neighborhoods have no specific meaning, but serve as a quick shortcut for routing to the right area.
+
+- For regional currencies, use a continent code from the folowing table:
```

Comment:
I do think `crypto` is a good top-level prefix, because there is actually a community around it. For large banks, it could make sense to use a currency/country code, or maybe to have some consortium of banks be the top-level prefix. Of course, this is less important for really big ledgers, because almost all connectors will have them in their routing tables.

Any ledger that isn't drawing a ton of traffic on its own will want to be addressed in terms of other ledgers. The current network of ILP early adopters might have its own top-level prefix, given that there's nothing to route in terms of yet.